shared/build.gradle.kts

```diff
@@ -23,7 +25,7 @@ kotlin {
 	sourceSets {
 		commonMain.dependencies {
 			implementation(libs.coroutines)
-			api(libs.firebase.auth)
+			implementation(libs.firebase.auth)
```

Comment:
That's the most important part, `auth` dependency is now encapsulated in `shared` module